insideout10/wordlift-plugin
# Feature/468 add schema.org website markup

Fixes #468 

#### Changes proposed in this Pull Request:
 * Change schema.org to [WebSite](http://schema.org/WebSite) for home and blog pages

Currently home page is uses schema.org/Article, while there is a schema.org/WebSite that can be used. In this PR there are two conditions that check if the home or blog page is being displayed and change the schema to WebSIte. It will also add default [Search Action].(https://developers.google.com/search/docs/data-types/sitelinks-searchbox)

There are also two filters (both documented in the code):
* `wordlift_disable_website_json_ld ` that allow you to disable the WebSite markup in case other plugin provides it.
* `wordlift_json_ld_search_url ` that allow you to change the default search url

#### Testing instructions for homepage:

* Activate the plugin and add your API key
* Create sample page
* Go to Settings > Reading and make that page homepage (Front page)
* Open the page and make sure that `application/ld+json` script has been changed to WebSite

#### Testing instructions for blogpage:

* Activate the plugin and add your API key
* Create sample page
* Go to Settings > Reading and make that page blog page (Posts page)
* Open the page and make sure that `application/ld+json` script has been changed to WebSite

Currently there are some errors related to publisher, that will be better to be addressed in another PR, so we can keep this one small.

![website-schema-check](https://cldup.com/mvOCYBEckT-3000x3000.png)

There are also three unit tests added that covered new schema and the two filters.

PS: Feedback welcome
